core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -361,4 +364,37 @@ private static TableMetadata commit(TableOperations ops, UpdateTableRequest requ
 
     return ops.current();
   }
+
+  /**
+   * This is a very simplistic approach that only validates the requirements for each table and does
+   * not do any other conflict detection. Therefore, it does not guarantee true transactional
+   * atomicity, which is left to the implementation details of a REST server.
+   */
+  public static void commitTransaction(Catalog catalog, CommitTransactionRequest request) {
```

Comment:
I think that this method should check whether the underlying catalog implements `commitTransaction`. If it does, then it should delegate to the catalog. If not, it should throw an exception.

The `CatalogHandlers` act as a reference implementation for the REST protocol, so I think it would be problematic if we included a method that took an atomic multi-table commit and implemented it as a sequence of individual commits.